### PR TITLE
Upgrade sdk and test / verify projects to net8; unquote itself stays at netstandard2.0

### PR DESCRIPTION
**Pull request by stephen-swensen**
See https://github.com/SwensenSoftware/unquote/issues/167

### CODE REVIEW

**Comment by bartelink** [comment]
TL;DR could the FSharp.Core dep be lowered back down to 6.0.7 (or lower) please?

----

Updating the FSharp.Core causes a lot of run on implications
For example: https://github.com/fsprojects/Argu/pull/237
This is not dissimilar to how FsHttp also did a similar update: https://github.com/fsprojects/FsHttp/issues/183
[FSharp.AWS.DynamoDB](https://www.nuget.org/packages/fsharp.aws.dynamodb) and other low level components depend on unquote
TaskSeq follows this approach too: https://github.com/fsprojects/FSharp.Control.TaskSeq/pull/123

My personal stuff depends on 6.0.7 for two reasons
- there's a bug fixed in there where the process can be torn down without a stacktrace under Async.Parallel.
- I used Async.Parallel in an internal implementation

NOTE: I do get that there's no actual reason why some old thing from 10 yars ago can't 'Just' update it's FSharp.Core from 4.3.2 to 8.0.100 without expecting it to just work, but that's not the point - it should simply depend on the lowest version possible
